.github/workflows/main.yml

```diff
@@ -117,6 +126,143 @@ jobs:
           CC=arm-linux-gnueabihf-gcc $(go env GOPATH)/bin/gox -cgo -os=linux -arch=arm -output=$GOX_OUTPUT ./...
           CC=aarch64-linux-gnu-gcc   $(go env GOPATH)/bin/gox -cgo -os=linux -arch=arm64 -output=$GOX_OUTPUT ./...
 
+          mkdir -p test/amd64/linux/bin
+          go build -tags=all,tests -o test/amd64/linux/bin ./...
+          CC=arm-linux-gnueabihf-gcc $(go env GOPATH)/bin/gox -cgo -os=linux -arch=arm -output=$GOX_TEST_OUTPUT -tags=all ./...
+          CC=aarch64-linux-gnu-gcc   $(go env GOPATH)/bin/gox -cgo -os=linux -arch=arm64 -output=$GOX_TEST_OUTPUT -tags=all ./...
+
```

Comment:
Can drop upload artifacts . Can also drop computing the ziti version. That's done below in the publish task

Reply:
Forgot to remove that upload. The test does need ziti version, I had left in the second one in because since we we're rebuilding everything anyway just incase. Can remove the second one if you'd like